analyzers/src/SonarAnalyzer.Common/Rules/ClassShouldNotBeEmptyBase.cs

```diff
@@ -44,15 +51,20 @@ public abstract class ClassShouldNotBeEmptyBase<TSyntaxKind> : SonarDiagnosticAn
             {
                 if (Language.Syntax.NodeIdentifier(c.Node) is { IsMissing: false } identifier
                     && IsEmptyAndNotPartial(c.Node)
-                    && !ShouldIgnoreBecauseOfBaseClass(c.Node, c.SemanticModel))
+                    && !HasAnyAttribute(c.Node)
+                    && !HasConditionalCompilationDirectives(c.Node)
+                    && !ShouldIgnoreBecauseOfBaseClassOrInterface(c.Node, c.SemanticModel))
                 {
                     c.ReportIssue(Diagnostic.Create(Rule, identifier.GetLocation(), DeclarationTypeKeyword(c.Node)));
                 }
             },
             Language.SyntaxKind.ClassAndRecordClassDeclarations);
 
-    private bool ShouldIgnoreBecauseOfBaseClass(SyntaxNode node, SemanticModel model) =>
+    private bool ShouldIgnoreBecauseOfBaseClassOrInterface(SyntaxNode node, SemanticModel model) =>
         IsClassWithDeclaredBaseClass(node)
-        && model.GetDeclaredSymbol(node) is INamedTypeSymbol classSymbol
-        && (classSymbol.BaseType is { IsAbstract: true } || classSymbol.DerivesFromAny(BaseClassesToIgnore));
+        && (HasGenericBaseClassOrInterface(node)
```

Comment:
IMO `HasGenericBaseClassOrInterface(node)` can be moved outside of this method (at line 55), because it doesn't need the semantic model and anyway it will again checks `Inherits.Count: > 0` so  it's safe

Like this you separate concerns.

Reply:
It doesn't need the semantic model, but I felt it belongs there because we're checking the base class/interface in this method, and that's exactly what `HasGenericBaseClassOrInterface` does.